### PR TITLE
DR-715 Integrate allure reporter

### PR DESCRIPTION
**Pull request by solomon209**
## Description

This work integrates allure report into the uec-cm UI framework

## Context

To provide test run feedback in a graphical report

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/ui/dist/reporter.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const allure_cucumberjs_1 = require("allure-cucumberjs");
+class Reporter extends allure_cucumberjs_1.CucumberJSAllureFormatter {
+    constructor(options) {
+        super(options, new allure_cucumberjs_1.AllureRuntime({
+            resultsDir: "./allure-results"
+        }), {
+            exceptionFormatter: (message) => message.replace("qwerty123", "[password edited]"),
+            labels: [
```
Comment (tamaragoldschmidt) [inline tests/ui/dist/reporter.js:10]: why is there a replace qwerty123 here?

File: tests/ui/package.json [hunk 2]
```diff
@@ -9,11 +8,20 @@
...
   },
   "scripts": {
+    "build": "tsc -p tsconfig-build.json",
     "pretest": "npx ts-node src/utilities/init.ts",
-    "test": "cross-env test_env=dev cucumber-js test",
-    "test_cmd_line": "cucumber-js test"
+    "test": "npm run build && cross-env test_env=dev cucumber-js test",
+    "tests:tagged": "node ./node_modules/.bin/cucumber-js  --tags",
+    "test_cmd_line": "cucumber-js test",
+    "allure": "npx allure generate allure-results -c -o allure-reports",
```
Comment (tamaragoldschmidt) [inline tests/ui/package.json:24]: i thought that we were going for the single file version of the report. That would remove the need to serve and allow for the report to be share more easily

File: tests/ui/package.json [hunk 3]
```diff
@@ -9,11 +8,20 @@
...
-    "@playwright/test": "^1.41.2"
+    "@cucumber/cucumber": "^10.3.1",
+    "@playwright/test": "^1.41.2",
+    "allure-commandline": "^2.27.0",
+    "allure-cucumberjs": "^2.12.0",
+    "allure-js-commons": "^2.12.0",
+    "allure-playwright": "^2.12.0"
   },
   "scripts": {
+    "build": "tsc -p tsconfig-build.json",
```
Comment (tamaragoldschmidt) [inline tests/ui/package.json:19]: Is there anything wrong with using the report.js file and not converting it?

Comment (solomon209) [reply tests/ui/package.json:19]: Nothing really. I assumed you wanted it as a .ts file as all other files are .ts.

File: tests/ui/dist/reporter.js (hunk 1 above)
Comment (solomon209) [reply tests/ui/dist/reporter.js:10]: Will remove

File: tests/ui/package.json (hunk 2 above)
Comment (solomon209) [reply tests/ui/package.json:24]: Fixed

**Pull request closed without merging**